rtl/eclrtl/rtlfield.cpp

```diff
@@ -2870,6 +3011,27 @@ int RtlSetTypeInfo::compare(const byte * left, const byte * right) const
     }
 }
 
+unsigned RtlSetTypeInfo::hash(const byte * self, unsigned inhash) const
+{
+    const bool allLeft = *(const bool *) self;
+    self += sizeof(bool);
+    if (allLeft)
+    {
+        // Nothing - this is unfortunate as it means hash(all) and hash([]) are the same...
+        // But it matches generated code
+    }
+    else
+    {
+        size32_t size = rtlReadSize32t(self);
+        self += sizeof(size32_t);
+        for (size32_t offset = 0; offset < size; offset += child->size(self + offset, self + offset))
```

Comment:
Rather than iterating the set, the generate code for a hash32 hashes it as a single block:

    vD = rtlHash32Data(vG,vF.getbytes(),2166136261U);

The code above is more correct, but probably better to be compatible, and change code generator at the same time if the semantics are changed.